### PR TITLE
VAULT007: guard max deposit/mint at zero rate

### PR DESCRIPTION
**Pull request by 0xferit**
## Summary
- return 0 from maxDeposit/maxMint when the exchange rate is zero
- add a yield skimming accounting regression test for the zero-rate edge case

## Testing
- forge test --match-path test/unit/strategies/yieldSkimming/Accounting.t.sol -vv

### CODE REVIEW

**Comment by 0xferit** [comment]
@codex review

File: test/unit/strategies/yieldSkimming/Accounting.t.sol [hunk 1]
```diff
@@ -599,6 +599,22 @@ contract AccountingTest is Setup {
...
+    function test_maxDeposit_maxMint_return_zero_when_rate_zero() public {
+        address alice = makeAddr("alice");
+        MockStrategySkimming(address(strategy)).updateExchangeRate(0);
+
+        assertEq(strategy.maxDeposit(alice), 0, "maxDeposit should be zero at rate 0");
+        assertEq(strategy.maxMint(alice), 0, "maxMint should be zero at rate 0");
+
+        vm.prank(alice);
+        vm.expectRevert();
+        strategy.deposit(1e18, alice);
```
[the remark this answers is not in the document]
Comment (0xferit) [reply test/unit/strategies/yieldSkimming/Accounting.t.sol:611]: Thanks! Keeping generic expectRevert intentionally: revert reason can vary (ZERO_SHARES vs div-by-zero) depending on rate math. The test’s goal is maxDeposit==0 and that deposit reverts at rate==0, not the exact error string.

File: test/unit/strategies/yieldSkimming/Accounting.t.sol [hunk 2]
```diff
@@ -599,6 +599,22 @@ contract AccountingTest is Setup {
...
+
+        vm.prank(alice);
+        vm.expectRevert();
+        strategy.mint(1e18, alice);
```
[the remark this answers is not in the document]
Comment (0xferit) [reply test/unit/strategies/yieldSkimming/Accounting.t.sol:615]: Same rationale here: keep expectRevert generic because exact revert depends on the underlying math (division by zero vs ZERO_SHARES). We’re asserting the invariant that maxMint==0 implies minting reverts at rate==0.

**Review by skimaharvey — APPROVED** [review]
Approving - while this is a very edge case that only applies when the vault is empty (unlikely scenario), it's still needed for ERC-4626 compliance. When the vault has deposits, zero rate already triggers insolvency which returns 0. This guard covers the empty vault edge case where insolvency check passes but deposits would still revert.
